contracts/HashRegistrarSimplified.sol

```diff
@@ -238,7 +242,38 @@ contract Registrar {
         }
         return len;
     }
+    
+    /** 
+     * @dev Creates a scheduled release for each name
+     * 
+     * Each name will be assigned a random date in which its auction 
+     * can be started, from 0 to 6 months
+     * 
+     * @param _hash The hash to start an auction on
+     * @param _timestamp The timestamp to query about
+     */
+     
+    function isAllowed(bytes32 _hash, uint _timestamp) constant returns (bool allowed){
+        uint upperLimit;
+        upperLimit--;       // Underflows to set it to 0xFFF...
+        uint timeSpan = _timestamp - registryStarted;
+        uint limit = timeSpan < launchLength ? 
+            timeSpan * (upperLimit / launchLength)
+            : upperLimit;
+        
+        return uint(_hash) < limit;
+    }
 
+    /** 
+     * @dev Returns expected launch date for hash
+     * 
+     * Returns the minimum date for opening the auction
+     * 
+     * @param _hash The hash to start an auction on
+     */
+    function getAllowedTime(bytes32 _hash) constant returns (uint timestamp) {
+        return registryStarted + launchLength*(uint(_hash)/2**128)/2**128;
```

Comment:
According to remix bitshifting is actually more expensive: 
```
getAllowedTime(bytes32): 449
getAllowedTimeBitshift(bytes32): 527  
```

But I'll let it this way because I learned something new and it's more elegant that way